frontend/sentry.client.config.ts

```diff
@@ -24,7 +25,7 @@ Sentry.init({
    },
    dsn: SENTRY_DSN,
    integrations: [new BrowserTracing()],
-   sampleRate: 1.0,
+   sampleRate: VERCEL_ENV === 'production' ? 0.05 : 1.0,
```

Comment:
Should we only report in dev?

That's what we currently do in the mono repo, but I am unsure if there is enough of a reason to keep reporting in non-prod envs


```suggestion
   sampleRate: VERCEL_ENV === 'production' ? 0.05 : 0,
```

Reply:
It's convenient to have error reporting turned on by default for QA and debugging/iterating. However, if we're worried about chipping away at our Sentry budget, I don't think QA would mind opting into error reporting like we do on the PHP side as long as we notified everyone about it.

Reply:
I think it's weird that local dev errors make it to Sentry and even cause Sentry's new PR feature to misreport errors:

Example:
https://github.com/iFixit/react-commerce/pull/1976#issuecomment-1730229461


https://ifixit.sentry.io/issues/4494886521/?project=6475315&referrer=github-pr-bot

Background:
https://github.com/getsentry/sentry/discussions/49996#discussioncomment-6807526

Reply:
I [see](https://ifixit.sentry.io/settings/billing/overview/) our quota is 50k, and 4k is a significant chunk of that. I think turning off reporting in dev by default makes sense 👍🏻

Reply:
See: https://ifixit.sentry.io/dashboard/65897/?project=6475315&statsPeriod=30d

I'd say more than half of all errors are from local dev.

The query there is looking at URL like `localhost`

Reply:
I agree. I almost *never* want to see dev errors and I practically only look at sentry to see prod errors. Seeing dev errors makes me generally less trusting of the stuff in sentry cause I'm constantly questioning... oh wait... did these happen in dev?

Reply:
Maybe we should take in a `SAMPLE_RATE variable and have it default to `0`? Then in vercel env conf we can bump it for just the `main` branch deployments.

Reply:
Sounds like a great idea to me